xbmc/PlayListPlayer.cpp

```diff
@@ -279,6 +280,9 @@ bool CPlayListPlayer::Play(int iSong, std::string player, bool bAutoPlay /* = fa
 
   m_iCurrentSong = iSong;
   CFileItemPtr item = playlist[m_iCurrentSong];
+  if (item->IsVideoDb() && !item->HasVideoInfoTag())
```

Comment:
I'm not sure this kind of logic should be something the PlaylistPlayer has to care about. It's IMO not within it's domain. Can't this be handled elsewhere?